tracer/tracer.go

```diff
@@ -501,7 +503,7 @@ func initializeMapsAndPrograms(kernelSymbols *libpf.SymbolMap, cfg *Config) (
 	}
 
 	if err = loadSystemConfig(coll, ebpfMaps, kernelSymbols, cfg.IncludeTracers,
-		cfg.OffCPUThreshold, cfg.FilterErrorFrames); err != nil {
+		cfg.OffCPUThreshold, cfg.FilterErrorFrames, cfg.MaxTailCalls); err != nil {
 		return nil, nil, fmt.Errorf("failed to load system config: %v", err)
```

Comment:
How about setting here, or even at `loadSystemConfig` a meaningful default for `MaxTailCalls` if its zero? Perhaps also add a comment that 29 is chosen to work with older kernels.

We also have somewhere code to detect running kernel, so perhaps this could be even autotuned based on detected kernel if left unset?